src/handlers.cpp

```diff
@@ -329,19 +329,20 @@ void ImpiTask::send_mar()
 
 void ImpiTask::on_mar_response(Diameter::Message& rsp)
 {
-  Cx::MultimediaAuthAnswer maa(rsp);
+  Cx::MultimediaAuthAnswer *maa = new Cx::MultimediaAuthAnswer(rsp);
```

Comment:
I did consider this, but the MAA can't be created until we've got the MAR response, so the MAA object would have be "undefined" until we get to this point.  Is there a C++ construct that lets you do this?